packages/react/src/components/Combobox/Combobox.test.tsx

```diff
@@ -0,0 +1,914 @@
+
+  fireArrowDownKeyPress();
+  assertOptionIsActive(0);
+  fireArrowDownKeyPress();
+  assertOptionIsActive(1);
+  fireArrowDownKeyPress();
+  assertOptionIsActive(2);
+});
+
+test('should prevent default event on home/end keypress', () => {
+  const preventDefault = spy();
+  render(
+    <Combobox label="label">
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+
+  const combobox = screen.getByRole('combobox');
+  const fireKeydownEvent = (key: string) => {
+    const event = createEvent.keyDown(combobox, { key });
+    // rtl doesn't let us mock preventDefault
+    // see: https://github.com/testing-library/react-testing-library/issues/572
+    event.preventDefault = preventDefault;
+    fireEvent(combobox, event);
+  };
+  expect(preventDefault.notCalled).toBeTruthy();
+  fireKeydownEvent('Home');
+  expect(preventDefault.callCount).toEqual(1);
+  fireKeydownEvent('End');
+  expect(preventDefault.callCount).toEqual(2);
+  fireKeydownEvent('ArrowDown');
+  expect(preventDefault.callCount).toEqual(2);
+});
+
+test('should call onActiveChange when active option changes', () => {
+  const onActiveChange = spy();
+  render(
+    <Combobox label="label" onActiveChange={onActiveChange}>
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+  const combobox = screen.getByRole('combobox');
+  // Note: Combobox forwards events to Listbox via dispatchEvent, but this doesn't
+  // work correctly within jsdom so we fire the events directly on listbox
+  const fireArrowDownKeyPress = () =>
+    fireEvent.keyDown(screen.getByRole('listbox'), { key: 'ArrowDown' });
+
+  fireEvent.focus(combobox);
+  assertListboxIsOpen(true);
+  expect(onActiveChange.notCalled).toBeTruthy();
+  fireArrowDownKeyPress();
+  expect(onActiveChange.callCount).toEqual(1);
+  expect(onActiveChange.getCall(0).firstArg.value).toEqual('Apple');
+  fireArrowDownKeyPress();
+  expect(onActiveChange.callCount).toEqual(2);
+  expect(onActiveChange.getCall(1).firstArg.value).toEqual('Banana');
+  fireArrowDownKeyPress();
+  expect(onActiveChange.callCount).toEqual(3);
+  expect(onActiveChange.getCall(2).firstArg.value).toEqual('Cantaloupe');
+});
+
+test('should set input value to empty string on open with selected option', () => {
+  render(
+    <Combobox label="label">
+      <ComboboxOption>Apple</ComboboxOption>
+      <ComboboxOption>Banana</ComboboxOption>
+      <ComboboxOption>Cantaloupe</ComboboxOption>
+    </Combobox>
+  );
+  const combobox = screen.getByRole('combobox');
+
+  fireEvent.focus(combobox);
+  assertListboxIsOpen(true);
+  fireEvent.click(screen.getAllByRole('option')[1]);
+  fireEvent.blur(combobox);
+  assertListboxIsOpen(false);
+  fireEvent.focus(combobox);
+  expect(screen.getByRole('combobox').getAttribute('value')).toEqual('');
```

Comment:
You might want to consider using `getByDisplayValue`: https://testing-library.com/docs/queries/bydisplayvalue/

Reply:
Is that actually what I want though? I want to assert that the combobox has an empty value.

`screen.getByDisplayValue('')` doesn't quite seem like the right assertion here, unless you were thinking of handling that differently somehow. I also don't necessarily want to make a negative assertion, i.e. `expect(screen.getByDisplayValue('banana')).isFalsy()`

Reply:
This one probably isn't the best example. I was more thinking of other places where you do things like this:

```
expect(screen.getByRole('combobox').getAttribute('value')).toEqual('Banana');
```

Reply:
I'm not following, and I'm not sure I agree that's the assertion that we want to make.

`screen.getByDisplayValue('Banana')` would fail if no input had that value, but I want a more specific assertion here since there's potentially multiple input fields with the visible + hidden field. I want a tighter assertion because I do think we want to validate that there is a `role="combobox"` field that has the matching field value. `getByDisplayValue` only solves half of that, unless we were to do something like `expect(screen.getByDisplayValue('banana').getAttribute('role')).toEqual('combobox')` which is the inverse of what we have above.

Reply:
Gotcha. That makes sense. The (mild) concern I have with using "getAttribute" is that it doesn't necessarily match what the user sees, which is the advantage of using `getByDisplayValue`.

It looks like cauldron is using @testing-library/jest-dom, so an alternative would be to use [toHaveDisplayValue](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tohavedisplayvalue):

```ts
expect(screen.getByRole('combobox')).toHaveDisplayValue('Banana');
```

Reply:
Yup, that's a fair assessment. I'll make that change.